### ISSUE TITLE
Integer ranges not checked correctly

### ISSUE THREAD

**Issue by sebastianbergmann** [vimeo/psalm#8838]
I have a method `Position::fromFileAndRank()` that uses `@psalm-param int<1,8>` annotations to limit the range of integer parameters to values >= 1 && <= 8:

```php
final class Position
{
    /**
     * @psalm-param int<1,8> $file
     * @psalm-param int<1,8> $rank
     */
    public static function fromFileAndRank(int $file, int $rank): self
    {
        return new self($file, $rank);
    }
}
```

I have code that calls the method shown above after ensuring that the arguments to be passed are within the range the parameters expect:

```php
$file = $this->position()->file() + $offset['file'];
$rank = $this->position()->rank() + $offset['rank'];

if ($file >= 1 && $file <= 8 && $rank >= 1 && $rank <= 8) {
    $positions[] = Position::fromFileAndRank($file, $rank);
}
```

But Psalm claims that the arguments are not within the range the parameters expect:

```
ERROR: [InvalidArgument] ... Argument 1 of Position::fromFileAndRank expects int<1, 8>, but int<min, max> provided

ERROR: [InvalidArgument] ... Argument 2 of Position::fromFileAndRank expects int<1, 8>, but int<min, max> provided
```

Reproducing example: https://psalm.dev/r/75a668a862

From the [documentation](https://psalm.dev/docs/annotating_code/type_syntax/scalar_types/#int-range) of integer ranges it is not clear to me whether Psalm only allows literal integers or whether it should be able to infer that a variable is within range.

**Comment by still-dreaming-1** [comment]
Interestingly, this workaround avoids the issue somehow: https://psalm.dev/r/000b9b94f9

**Comment by theodorejb** [comment]
@orklah It looks like this was also fixed by #8934.
